packages/ember-data/lib/system/record-arrays/record-array.js

```diff
@@ -95,6 +95,47 @@ export default Ember.ArrayProxy.extend(Ember.Evented, {
   },
 
   /**
+    Get a filtered subset of the underlying `RecordArray`.
+    The subset updates when a record would match or mismatch the
+    specified filter parameters.
+
+    Example
+
+    ```javascript
+    var allToms = store.all('person').filterBy('name', 'Tom');
+
+    allToms.get('length'); // 0, since no toms yet in store
+
+    var tom = store.push('person', { id: 1, name: 'Tom' });
+    allToms.get('length'); // Tom is added
+
+    tom.set('name', 'Thomas');
+    allToms.get('length'); // 0, since no more records with name === 'Tom'
+    ```
+
+    @method filterBy
+    @param {String} key property path
+    @param {*} value optional
+
+  */
+  filterBy: function(key, value) {
+    var content = get(this, 'content');
+
+    var FilteredRecordArray = Ember.ArrayProxy.extend({
+      // since content is an array of InternalModel's and we need to observe
+      // the property at path `key`, the records are mapped to
+      // `internalRecords`, so we can use the `Ember.computed.filterBy` macro,
+      // which listens to changes of this property
+      internalRecords: Ember.computed.mapBy('content', 'record'),
+      arrangedContent: Ember.computed.filterBy('internalRecords', key, value)
```

Comment:
I'm very unsure about this implementation, since `record` on the `InternalModel` should be accessed via [`getRecord()`](http://git.io/vIEk3) I guess...